### PR TITLE
WIP: ScaleOverlayTool: Basic code

### PR DESCRIPTION
**Pull request by galelis**

### CODE REVIEW

**Comment by galelis** [comment]
TODO's:

Still missing unit tests.
Confirm with Danny if onDisabled and onEnabled callbacks should be create as I did

File: examples/00-scaleOverlay/index.html [hunk 1]
```diff
@@ -0,0 +1,121 @@
+<!DOCTYPE HTML>
```
Comment (dannyrb) [inline examples/00-scaleOverlay/index.html]: I'm a little torn. On one-hand simple, granular examples are easier to learn from to use a tool. On the other hand, a lot of simple examples can become hard to maintain. For the time being, until we get things hammered out, I think we should slowly add on to the "global" example, and delete the ones that are no longer applicable.

When this becomes the official branch, we can double back and rethink how best to create and maintain examples as a form of documentation.

Comment (dannyrb) [reply examples/00-scaleOverlay/index.html]: If you'd like, we can merge with this, and I can use it as a guide to update the primary example, and then delete it? I've also wanted to add a simple way to bind to different mouse buttons in the main example so you can test multiple tools in tandem.

File: src/fancy-tools/scaleOverlay.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
+import external from './../externalModules.js';
+import baseTool from './../base/baseTool.js';
+// Drawing
+import { getNewContext, draw, path } from '../util/drawing.js';
+
+const cornerstone = external.cornerstone;
+
+const drawLine = (context, startPoint, endPoint) => {
```
Comment (dannyrb) [inline src/fancy-tools/scaleOverlay.js]: Can we use the `drawLine` from `util/drawing.js` instead

File: src/fancy-tools/scaleOverlay.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    });
+  }
+
+  enabledCallback (element) {
+    external.cornerstone.updateImage(element);
+  }
+
+  disabledCallback (element) {
+    external.cornerstone.updateImage(element);
+  }
```
Comment (dannyrb) [inline src/fancy-tools/scaleOverlay.js]: `enabledCallback` and `disabledCallback` look good 👍 

each `setToolMode` method calls a callback with the following formula:
- `tool['${mode}Callback'](element, options);`

File: src/fancy-tools/scaleOverlay.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+export default class extends baseTool {
+  constructor (name) {
+    super({
+      name: name || 'scaleOverlay',
+      configuration: {
+        color: 'white',
+        lineWidth: 2,
+        shadowColor: 'black',
+        shadowBlur: 4,
```
Comment (dannyrb) [inline src/fancy-tools/scaleOverlay.js]: We might want to keep track of what common properties we may want to have a global setting/default for, and then allow a local value to override. That way it's easier to set values that affect all tools, and override them as needed.

**Comment by dannyrb** [comment]
Example URL: https://deploy-preview-509--cornerstone-tools.netlify.com/examples/00-scaleoverlay/

File: src/fancy-tools/scaleOverlay.js (hunk 4 above)
Comment (galelis) [reply src/fancy-tools/scaleOverlay.js]: You mean I should be using the `stateManagement/` to get those values?

File: src/fancy-tools/scaleOverlay.js (hunk 2 above)
Comment (galelis) [reply src/fancy-tools/scaleOverlay.js]: yeah, I finished this first part, with the idea of just make it work. 
I will refactor :)

File: examples/00-scaleOverlay/index.html (hunk 1 above)
Comment (galelis) [reply examples/00-scaleOverlay/index.html]: hmm, make sense. 
I will figure out a way to add this tool into the global example and delete this example.

**Comment by dannyrb** [comment]
image

**Pull request merged**